juju/juju
# Remove noble from supported versions in 2.9

Noble was erroneously added as a supported version in 2.9. I believe some manual checks may have passed due to the type in the base.

PR #17283 which purported to stop 2.9 supporting noble in fact only updated some of the necessary functions. It turns out that there are two ways of using the os/series package to get the info from distroinfo. Once of them is to use `series.LocalSeriesVersionInfo` which was fixed in this PR and the other is by getting `series.NewDistroInfo` and updating the supported series by reading from that.

This method loops through the `ubuntuSeries` and checks if any of the information in it has changed in `NewDistroInfo`. Therefore, to stop supporting noble in this `ubuntuSeries` map we only need to remove noble from this list (to which it should never have been added in the first place). 

In #17283 some tests were modified to include noble, these changes are also reversed.

## Checklist




## QA steps
Go to `/testcharms/charm-hub/charms/juju-qa-test`
Add the following to the charmcraft.yaml:
```
bases:
    - build-on:
        - name: "ubuntu"
          channel: "24.04"
      run-on:
        - name: "ubuntu"
          channel: "24.04"
          architectures: ["amd64", "arm64"]
```
Then do:
```
charmcraft pack
juju bootstrap lxd
juju deploy ./juju-qa-test_ubuntu-24.04-amd64-arm64.charm --series noble
```
Observe error.
